### DISCUSSION TITLE
fish in far2l

### DISCUSSION THREAD

**Discussion by AlexS54** [elfmz/far2l#2268] [Q&A] [answered]
Hello
I can't seem to make fish work inside far2l (instead of the default bash), on macOS
If I change the "Use shell" setting to the fish path, there's simply no output for any command I try in the embedded terminal.
Any quick tutorial on how to do this?

For context, both `fish` and `far2l` are installed with homebrew, and I'm running macOS Sonoma Version 14.5 (23F79)

**Comment by akruphi** [comment, marked as answer]
FAR2L internal command line work fully only via **bash**.

You can change shell by **Menu(F9)**->**Options**->**Command line settings**->**Use shell** but command line will work with significant restrictions/bugs especially with native shell commands.

Explanation: https://github.com/elfmz/far2l/issues/1264#issuecomment-1045334082
